maven/impl-maven/src/main/java/org/jboss/shrinkwrap/resolver/impl/maven/bootstrap/MavenSettingsBuilder.java

```diff
@@ -188,9 +188,9 @@ public Settings buildSettings(SettingsBuildingRequest request) {
 
         // get settings object and update it according to property values
         Settings settings = result.getEffectiveSettings();
-        settings = enrichWithLocalRepository(settings);
-        settings = enrichWithOfflineMode(settings);
-        settings = decryptPasswords(settings);
+        enrichWithLocalRepository(settings);
+        enrichWithOfflineMode(settings);
+        decryptPasswords(settings);
```

Comment:
I'm almost sure you're loosing state (if the methods didn't change since this code was written)